sale_coupon_advanced_option/models/sale_coupon_program.py

```diff
@@ -0,0 +1,52 @@
+# Copyright 2020 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html)
+
+from odoo import _, api, fields, models
+
+
+class SaleCouponProgram(models.Model):
+    _inherit = "sale.coupon.program"
+
+    # Add possibility to use discount only on first order of a customer
+    first_order_only = fields.Boolean(
+        string="Apply only first",
+        default=False,
+        help="Apply only on the first order of each client matching the conditions",
+    )
+
+    def _check_promo_code(self, order, coupon_code):
+        if (
+            self.first_order_only
+            and len(
+                self.env["sale.order"].search(
+                    [
+                        ("partner_id", "=", order.partner_id.id),
+                        ("state", "!=", "cancel"),
+                    ]
+                )
+            )
+            > 1
+        ):
+            return {"error": _("Coupon can be used only for the first sale order!")}
+        else:
+            return super()._check_promo_code(order, coupon_code)
+
+    def _filter_programs_by_sale_order_count(self, order):
+        if (
+            self.env["sale.order"].search_count(
+                [("partner_id", "=", order.partner_id.id), ("state", "!=", "cancel")]
+            )
+            <= 1
+        ):
+            return self.filtered(lambda program: program.first_order_only is True)
```

Comment:
The value is already a boolean, you don't need to test it against another boolean value
```suggestion
            return self.filtered(lambda program: program.first_order_only)
```

Reply:
@p-tombez, code is fixed.
regarding why it is not OCA, that was @gurneyalex  decision as we all functionalities in one feature/branch